lib/FlagUsingEvalPlugin.js

```diff
@@ -0,0 +1,40 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Sergey Melyukov @smelukov
+*/
+
+"use strict";
+
+/** @typedef {import("./Compiler")} Compiler */
+
+class FlagUsingEvalPlugin {
+	/**
+	 * @param {Compiler} compiler webpack compiler
+	 * @returns {void}
+	 */
+	apply(compiler) {
+		compiler.hooks.compilation.tap(
+			"FlagUsingEvalPlugin",
+			(compilation, { normalModuleFactory }) => {
+				const handler = parser => {
+					parser.hooks.call.for("eval").tap("FlagUsingEvalPlugin", () => {
+						parser.state.module.buildMeta.moduleConcatenationBailout = "eval()";
+						parser.state.module.buildMeta.usingEval = true;
```

Comment:
`buildMeta` should be `buildInfo` instead. This was already wrong in the original code.

`buildMeta` is for externally visible meta info, `buildInfo` is for internal meta info.

Since using `eval()` has no influence on consumers of a module it should be in `buildInfo`.